jubatus/core/driver/recommender.cpp

```diff
@@ -102,6 +103,40 @@ std::vector<std::pair<std::string, float> > recommender::similar_row_from_id(
 }
 
 std::vector<std::pair<std::string, float> >
+recommender::similar_row_from_id_and_score(
+    const std::string& id,
+    float score) {
+  std::vector<std::string> rows = get_all_rows();
+  size_t ret_num = rows.size();
+
+  std::vector<std::pair<std::string, float> > ret;
+  recommender_->similar_row(id, ret, ret_num);
+
+  std::vector<std::pair<std::string, float> >::iterator it = ret.begin();
+  while (it != ret.end()) {
+      if (it->second < score) {
+          it = ret.erase(it);
+      } else {
+          ++it;
+      }
+  }
+
+  return ret;
+}
+
+std::vector<std::pair<std::string, float> >
+recommender::similar_row_from_id_and_rate(
+    const std::string& id,
+    float rate) {
+  std::vector<std::string> rows = get_all_rows();
+  size_t ret_num = std::ceil(rows.size() * rate / 100);
```

Comment:
I think `0<= rate <= 1.0` is better because 0 to 1 rate is also used commonly and we can remove this division.

Reply:
👍 I think so.
I fixed.